nemo/collections/llm/api.py

```diff
@@ -247,6 +248,103 @@ def validate(
+        awq_block_size=awq_block_size,
+        sq_alpha=sq_alpha,
+        enable_kv_cache=enable_kv_cache,
+    )
+
+
+@run.cli.factory
+@run.autoconvert
+def default_export(
+    path: str,
+    dtype: Union[str, int] = "bf16",
+    decoder_type: Optional[str] = None,
+    inference_tensor_parallel: int = 1,
+    inference_pipeline_parallel: int = 1,
+) -> ExportConfig:
+    """Default export configuration."""
+    return ExportConfig(
+        path=path,
+        dtype=dtype,
+        decoder_type=decoder_type,
+        inference_tensor_parallel=inference_tensor_parallel,
+        inference_pipeline_parallel=inference_pipeline_parallel,
+    )
+
+
+@run.cli.entrypoint(name="ptq", namespace="llm")
+def ptq(
+    nemo_checkpoint: str,
+    # TODO: Maybe also create calibration_config for parallel and data settings?
+    calib_tp: int = 1,
+    calib_pp: int = 1,
+    dataset_size: int = 512,
+    batch_size: int = 64,
+    seq_len: int = 128,
+    quantization_config: QuantizationConfig = default_quantization(),
+    export_config: ExportConfig = default_export(None),
+):
+    """
+    Applies Post-Training Quantization (PTQ) for a model using the specified quantization and export configs. It runs
+    calibration for a small dataset to collect scaling factors low-precision GEMMs used by desired quantization method.
+
+    This function produces TensorRT-LLM checkpoint ready for deployment using nemo.export and nemo.deploy modules
+    or direcly using TensorRT-LLM library.
+
+    Args:
+        nemo_checkpoint (str): The path to model to be quantized.
+        calib_tp (int): Calibration tensor parallelism.
+        calib_pp (int): Calibration pipeline parallelism.
+        dataset_size (int): Number of samples to run calibration.
+        batch_size (int): Batch size for calibration.
+        seq_len (int): Length of calibration samples (in tokens).
+        quantization_config (QuantizationConfig): Configuration for quantization algorithm.
+        export_config (ExportConfig): Export configuration for TensorRT-LLM checkpoint.
+
+    Returns:
+        Path: The directory path where quantized model is saved.
+
+    Example:
+        >>> from nemo.collections.llm.quantization import ExportConfig, QuantizationConfig
+        >>> nemo_checkpoint = "/opt/checkpoints/LLAMA3-8B-fp16"
+        >>> quantization_config = QuantizationConfig(algorithm="fp8")
+        >>> export_config = ExportConfig(path="/opt/checkpoints/LLAMA3-8B-fp8")
+        >>> llm.ptq(nemo_checkpoint, quantization_config=quantization_config, export_config=export_config)
+        '/opt/checkpoints/LLAMA3-8B-fp8'
+    """
+    from nemo.collections.llm import quantization
+
+    quantizer = quantization.Quantizer(quantization_config, export_config)
+    model = quantization.load_with_modelopt_layer_spec(nemo_checkpoint, calib_tp, calib_pp)
+
+    get_dataloader = create_data_iterator_getter(...)  # TODO: need to import from somewhere, can't from scripts/llm/
```

Comment:
I'll need to move `create_data_iterator_getter` somewhere else from [scripts/llm/ptq.py](https://github.com/NVIDIA/NeMo/blob/371b4221e600056619b63b80f663f8dde4095a79/scripts/llm/ptq.py#L73) in https://github.com/NVIDIA/NeMo/pull/10642 so that we can import it here.